CharlesRajendran/product-api
# All Functional and Most Non Functional Requirements are Completed

# Description
- Features implemented
  - Implemented REST API's for CRUD Product
  - Bulk Insert with CSV
  - GraphQL Operations
  - Pagination and Sorting for GET API's
  - API Documentation
  - Seeding and Migration
  - Data Modeling
  - Redis cache refactoring
  - Docker Refactoring
  - Security Middleware Inclusions

- Fixes # (issue):
  - #8 
  - #9 
  - #10 
  - #11 
  - #12 
  - #15 
  - #16 
  - #17 
  - #18 
  - #19 

## Type of change

Please delete options that are not relevant.

- [ ] Other - Please make sure you explain what sort of change this is

# How to test this PR?

- Run the application with `npm run launch` (make sure you have docker installed)
- REST API Testing => http://localhost:3000/api-docs
- GraphQP Test => http://localhost:3000/graphql

# Checklist:
